src/components/table/data-table.tsx

```diff
@@ -42,8 +44,10 @@ export function DataTable<TData, TValue>({
   columns,
   data,
   pagination,
+  onSelectionChange,
 }: DataTableProps<TData, TValue>) {
   const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const [rowSelection, setRowSelection] = useState({});
```

Comment:
I'd suggest to move the whole state up to the use-cases/traces as that is the top-most component that can then pass down state to data-table and the action. This reduces the number of state replications that we do on each change.

Reply:
@marcklingen  Kindly guide me in issue that i have explained in following video.

Loome video:  https://www.loom.com/share/a50cb686c665461499a1f1ca700aeb6c?sid=3ce1c373-4e8e-4fb4-98e2-449ae9e51714

Reply:
I have resolved all issues except this one.

Reply:
Thanks, having a look

Reply:
solved this one